hub/core/query/filter.py

```diff
@@ -82,8 +111,27 @@ def filter_inplace(
 
         it = tqdm(it, total=len(dataset))
 
-    for i, sample_in in it:
-        if filter_function(sample_in):
-            index_map.append(i)
+    query_id = str(uuid4())
```

Comment:
Query Id has to be a hash of the `dataset.url`, `query_text`  and current `commit_id`

Reply:
query text is going to be "UDF" for functions, we'll end up with the same hash for all functions sent for a particular commit, better to just randomly generate imo

Reply:
With hash you can check, if the query is actually was already executed on particular commit, fetch the results and return immediately. For 'UDF', we have to think what would be the best way to hash. Maybe pickle and hash byte stream ?